onlinejudge/service/atcoder.py

```diff
@@ -575,15 +575,15 @@ def _from_html(cls, html: bytes, *, problem: 'AtCoderProblem', session: Optional
         else:
             assert False
 
-        for memory_limit_prefix in ('メモリ制限: ', 'Memory Limit: '):
-            if memory_limit.startswith(memory_limit_prefix):
-                break
-        else:
-            assert False
-        if memory_limit.endswith(' KB'):
-            memory_limit_byte = int(float(utils.remove_suffix(utils.remove_prefix(memory_limit, memory_limit_prefix), ' KB')) * 1000)
-        elif memory_limit.endswith(' MB'):
-            memory_limit_byte = int(float(utils.remove_suffix(utils.remove_prefix(memory_limit, memory_limit_prefix), ' MB')) * 1000 * 1000)
+        parsed_memory_limit = re.search(r'^(メモリ制限|Memory Limit): ([0-9.]+) (KB|MB)', memory_limit)
```

Comment:
```suggestion
        # When login as the admin, a link is added after memory limit. See https://github.com/online-judge-tools/api-client/issues/90
        parsed_memory_limit = re.search(r'^(メモリ制限|Memory Limit): ([0-9.]+) (KB|MB)', memory_limit)
```

(nitpick) 「非自明な理由により正規表現を使わないと壊れる」なので一応コメントしておきたい